src/scheduled_tasks.rs

```diff
@@ -0,0 +1,85 @@
+// Scheduler, and trait for .seconds(), .minutes(), etc.
+use clokwerk::{Scheduler, TimeUnits};
+// Import week days and WeekDay
+use diesel::{sql_query, PgConnection, RunQueryDsl};
+use lemmy_db_queries::{source::activity::Activity_, DbPool};
+use lemmy_db_schema::source::activity::Activity;
+use log::info;
+use std::{thread, time::Duration};
+
+/// Schedules various cleanup tasks for lemmy in a background thread
+pub fn setup(pool: DbPool) {
+  let mut scheduler = Scheduler::new();
+
+  let conn = pool.get().unwrap();
+  active_counts(&conn);
+  reindex_aggregates_tables(&conn);
+  scheduler.every(1.hour()).run(move || {
+    active_counts(&conn);
+    reindex_aggregates_tables(&conn);
+  });
+
+  let conn = pool.get().unwrap();
+  clear_old_activities(&conn);
+  scheduler.every(1.weeks()).run(move || {
+    clear_old_activities(&conn);
+  });
```

Comment:
Running clearing out old activities every week (it deletes them older than 6 months)